src/lib/date-utils.ts

```
@@ -305,3 +282,158 @@
+  let intervalStart: UTCDate;
+  let intervalEnd: UTCDate;
+
+  switch (intervalType) {
+    case "day":
+      intervalStart = toUTCMidnight(date);
+      intervalEnd = addDays(intervalStart, 1);
+      break;
+
+    case "week":
+      // Ensure we start from the Sunday of the week containing the date
+      const startOfWeek = toUTCMidnight(date);
```

Biome (1.9.4)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Unsafe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)